base_exception/models/base_exception.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2011 Raphaël Valyi, Renato Lima, Guewen Baconnier, Sodexis
+# Copyright 2017 Akretion (http://www.akretion.com)
+# Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import time
+from functools import wraps
+from odoo import api, fields, models, _
+import logging
+
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+def implemented_by_base_exception(func):
+    """Call a prefixed function based on 'namespace'."""
+    @wraps(func)
+    def wrapper(cls, *args, **kwargs):
+        fun_name = func.__name__
+        fun = '_%s%s' % (cls.rule_group, fun_name)
+        if not hasattr(cls, fun):
+            fun = '_default%s' % (fun_name)
+        return getattr(cls, fun)(*args, **kwargs)
+    return wrapper
+
+
+class ExceptionRule(models.Model):
+    _name = 'exception.rule'
+    _description = "Exception Rules"
+    _order = 'active desc, sequence asc'
+
+    name = fields.Char('Exception Name', required=True, translate=True)
+    description = fields.Text('Description', translate=True)
```

Comment:
technically string is not necessary anymore when same as column name

Reply:
not agree :)